src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
+      throw new WalletRequestError('Error getting wallet history.');
     }
     return history
   }
 
+  /**
+   * Get utxo from tx id and index
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  async getUtxoFromId(txId: string, index: number): Promise<Utxo | null> {
+    const response = await walletApi.getUtxos(this, { txId, index });
+    if (response.status === 200 && response.data.success === true) {
+      const utxos = response.data.utxos;
+      if (utxos.length === 0) {
+        // No utxo for this txId/index or is not from the requested wallet
+        return null;
+      } else {
+        return utxos[0];
+      }
+    } else {
+      throw new WalletRequestError('Error requesting utxo.');
+    }
+  }
+
+  /**
+   * Get utxos for filling a transaction
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  async getUtxos(options: { tokenId?: string, authority?: number, addresses?: string[], totalAmount?: number, count?: number } = {}): Promise<{ utxos: Utxo[], changeAmount: number }> {
+    type optionsType = {
+      tokenId: string,
+      authority: number | null,
+      addresses: string[] | null,
+      totalAmount: number | null,
+      count: number,
+      ignoreLocked: boolean,
+    };
+    const newOptions: optionsType = Object.assign({
+      tokenId: HATHOR_TOKEN_CONFIG.uid,
+      authority: null,
+      addresses: null,
+      totalAmount: null,
+      count: 1,
+    }, options);
+
+    if (!newOptions.authority && !newOptions.totalAmount) {
+      throw new UtxoError('We need the total amount of utxos if it\'s not an authority request.');
+    }
+
+    newOptions['ignoreLocked'] = true;
+
+    const response = await walletApi.getUtxos(this, newOptions);
+    let changeAmount = 0;
+    let utxos: Utxo[] = []
+    if (response.status === 200 && response.data.success === true) {
+      const retUtxos = response.data.utxos;
+      if (retUtxos.length === 0) {
+        // No utxos available for the requested filter
+        utxos = retUtxos;
+      } else if (newOptions.authority) {
+        // Requests an authority utxo, then I return the count of requested authority utxos
+        utxos = retUtxos.slice(0, newOptions.count);
+      } else {
+        // We got an array of utxos, then we must check if there is enough amount to fill the totalAmount
+        // and slice the least possible utxos
+        const ret = transaction.selectUtxos(retUtxos, newOptions.totalAmount!);
+        changeAmount = ret.changeAmount;
+        utxos = ret.utxos;
+      }
+    } else {
+      throw new WalletRequestError('Error requesting utxo.');
+    }
+    return { utxos, changeAmount };
+  }
+
+  /**
+   * Calculate sign message for auth token
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  getAuthSign(timestamp: number): string {
+    const xpriv = wallet.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
```

Comment:
Why don't we get the xprivkey directly from the storage? I mean, do we need really to access the seed and passphrase?

Reply:
To get the private key from the storage we would need a bigger refactor in the wallets and/or in the auth methods. RIght now we depend on the xpriv to create the jwt auth and this expires in 30 minutes. We need this for every wallet service request and this refactor need to be discussed before being implemented. I believe this is not the best implementation about handling the xpriv but I think this should be handled in another PR

Reply:
I agree. Just make sure we have another issue to get it improved in the future.

Reply:
We have